internal/config/matcher.go

```diff
@@ -86,3 +84,23 @@ func (m *Matcher) ForRequest(tenant string, groups []string) *Matcher {
 
 	return m
 }
+
+func (m *Matcher) ViaQToOTELMigration(selectors map[string][]string) {
```

Comment:
This is currently just a "feeling" as I have not written a test for it yet or reproduced it in a cluster, but I think modifying `m.Keys` might affect future requests, because `Matcher.ForRequest()` does not actually produce a _copy_.

Reply:
Ahh nice catch! I've modify `ForRequest` to return a new Marcher instead of the original. I've also simplified ViaQToOTELMigration